### PR TITLE
Add logging to Daisy

### PR DESCRIPTION
**Pull request by adjackura**
- Add logging that prints to stdout and streams to GCS
- Rename Workflow.Bucket to GCSPath 
- Fix sources upload

### CODE REVIEW

File: daisy/workflow/workflow.go [hunk 1]
```diff
@@ -259,16 +274,39 @@ func (w *Workflow) FromFile(file string) error {
...
+
+	w.logger.Print("Validating workflow")
+	if err := w.validate(); err != nil {
+		w.logger.Printf("Error validating workflow: %v", err)
+		w.Cancel()
+		return err
+	}
+	w.logger.Print("Uploading sources")
+	if err := w.uploadSources(); err != nil {
+		w.logger.Printf("Error validating workflow: %v", err)
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: s/validating workflow/uploading workflow sources

Maybe?

File: daisy/workflow/workflow.go [hunk 2]
```diff
@@ -343,6 +381,27 @@ func (w *Workflow) nameToInstanceLink(n string) string {
 	return fmt.Sprintf("projects/%s/zones/%s/instances/%s", w.Project, w.Zone, n)
 }
 
+type gcsLogger struct {
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: I've been trying to keep things in alphabetical order and categorized by their associated structs. e.g. step methods with step, resource methods with resource. You're a savage and put gcsLogger right in the middle of the (w *Workflow) methods.

File: daisy/workflow/workflow.go [hunk 3]
```diff
@@ -524,29 +591,25 @@ func (w *Workflow) traverseDAG(f func(step) error) error {
...
+			src := w.StorageClient.Bucket(b).Object(o)
+			if _, err := dst.CopierFrom(src).Run(w.Ctx); err != nil {
 				return err
 			}
 		} else {
 			// Local to GCS.
-			writer := dstO.NewWriter(w.Ctx)
-			bs, err := ioutil.ReadFile(origPath)
+			gcs := dst.NewWriter(w.Ctx)
+			f, err := os.Open(origPath)
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: Cool, I was a little nervous about my bufio logic here. This seems simpler.

File: daisy/workflow/workflow.go (hunk 1 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: Done

File: daisy/workflow/workflow.go (hunk 2 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: :P I usually place functions right above where they are called to appease the entropy gods. Moved this to the top to meet you absurd ordering ideas

File: daisy/workflow/step_create_instances.go [hunk 4]
```diff
@@ -108,14 +108,14 @@ func (c *CreateInstances) run(w *Workflow) error {
 				default:
 					startup = "startup-script-url"
 				}
-				inst.AddMetadata(map[string]string{startup: path.Join(w.sourcesPath, ci.StartupScript)})
+				inst.AddMetadata(map[string]string{startup: "gs://" + path.Join(w.bucket, w.sourcesPath, ci.StartupScript)})
```
Comment (crunk1) [inline daisy/workflow/step_create_instances.go]: What does the "startup" key-value do? I thought GCS paths were "startup-script-url".

File: daisy/workflow/workflow.go [hunk 5]
```diff
@@ -393,10 +452,28 @@ func (w *Workflow) populate() error {
...
+		for parent := w.parent; parent != nil; parent = w.parent.parent {
+			name = parent.Name + "." + name
+		}
+		prefix := fmt.Sprintf("[%s]: ", name)
+		flags := log.Ldate | log.Ltime
+		log.New(os.Stdout, prefix, flags).Println("Logs will be streamed to", "gs://"+path.Join(w.bucket, w.logsPath, "daisy.log"))
+		w.logger = log.New(io.MultiWriter(os.Stdout, gcs), prefix, flags)
+	}
 
 	for name, step := range w.Steps {
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: Overloading "step".

s/step/s here?

File: daisy/workflow/step_create_instances.go (hunk 4 above)
Comment (adjackura) [inline daisy/workflow/step_create_instances.go]: I set that variable in the preceding switch as windows and linux are slightly different.

**Pull request merged**